pkg/tasks/thanos_ruler_user_workload.go

```diff
@@ -275,7 +312,7 @@ func (t *ThanosRulerUserWorkloadTask) destroy() error {
 			return errors.Wrap(err, "deleting Thanos Ruler trusted CA bundle ConfigMap failed")
 		}
 
-		tr, err := t.factory.ThanosRulerCustomResource(trustedCA)
+		tr, err := t.factory.ThanosRulerCustomResource(trustedCA, nil)
```

Comment:
this will immediately return with an error because no secret is given?

Reply:
I see, thats why tests are failing, makes sense!

Reply:
fixed, this should fix the failing tests as well, nicely spotted!